MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationOptionsActivity.java

```diff
@@ -0,0 +1,198 @@
+import com.mapbox.android.core.permissions.PermissionsListener;
+import com.mapbox.android.core.permissions.PermissionsManager;
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.plugins.locationlayer.LocationLayerOptions;
+import com.mapbox.mapboxsdk.plugins.locationlayer.LocationLayerPlugin;
+import com.mapbox.mapboxsdk.plugins.locationlayer.OnLocationLayerClickListener;
+import com.mapbox.mapboxsdk.plugins.locationlayer.modes.CameraMode;
+import com.mapbox.mapboxsdk.plugins.locationlayer.modes.RenderMode;
+
+import java.util.List;
+
+/**
+ * Use the LocationLayerOptions class to customize the Location Layer Plugin's
+ * device location icon.
+ */
+public class LocationOptionsActivity extends AppCompatActivity implements
+    OnMapReadyCallback, PermissionsListener, OnLocationLayerClickListener, LocationEngineListener {
+
+  private PermissionsManager permissionsManager;
+  private MapboxMap mapboxMap;
+  private MapView mapView;
+  private LocationEngine locationEngine;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_location_plugin_options);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    LocationOptionsActivity.this.mapboxMap = mapboxMap;
+    enableLocationPlugin();
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  private void enableLocationPlugin() {
+    // Check if permissions are enabled and if not request
+    if (PermissionsManager.areLocationPermissionsGranted(this)) {
+
+      // Set up the location engine
+      initializeLocationEngine();
+
+      // Create and customize the plugin's options
+      LocationLayerOptions locationLayerOptions = LocationLayerOptions.builder(this)
+          .elevation(5)
+          .accuracyAlpha(.6f)
+          .accuracyColor(Color.RED)
+          .foregroundDrawable(R.drawable.android_custom_location_icon)
+          .build();
+
+      // Create the plugin
+      LocationLayerPlugin locationLayerPlugin = new LocationLayerPlugin(
+          mapView, mapboxMap, locationEngine, locationLayerOptions);
+
+      // Set the plugin's camera and render modes
+      locationLayerPlugin.setCameraMode(CameraMode.TRACKING);
```

Comment:
What about adding a tracking listener and a button that lets you enable the tracking back whenever broken?

Reply:
Added ✔️